File: java/testsuite/Jogl4Examples/SolidTextureExample/src/options/CommandLineOptions.java
package options;

import model.SolidTextureModel;

public class CommandLineOptions {
    private final SolidTextureModel model;

    public CommandLineOptions(SolidTextureModel model) {
        this.model = model;
    }

    public void processArguments(String[] args) {
        if ( args == null ) {
            return;
        }

        for ( int i = 0; i < args.length; i++ ) {
            String arg = args[i];
            if ( "-tangibleServer".equals(arg) && i + 1 < args.length ) {
                model.setTangibleServiceUrl(args[++i]);
            }
        }
    }
}

File: cpp/testsuite/OpenGL4Examples/MeshExample/src/options/CommandLineOptions.cpp
#include "java/lang/String.h"
#include "model/MeshModel.h"
#include "options/CommandLineOptions.h"

CommandLineOptions::CommandLineOptions(MeshModel* model)
    : model(model)
{
}

void CommandLineOptions::processArguments(int argc, char** argv)
{
    if ( argv == 0 ) {
        return;
    }

    for ( int i = 0; i < argc; i++ ) {
        java::String arg(argv[i]);
        if ( arg == "-tangibleServer" && i + 1 < argc ) {
            model->setTangibleServiceUrl(java::String(argv[++i]));
        }
    }
}
